tap_pardot/streams.py

```diff
@@ -152,6 +153,32 @@ def get_params(self):
         }
 
 
+class CreatedAtReplicationStream(Stream):
+    """
+    Streams where records are immutable and can only be sorted by created_at.
+
+    If no config is provided, it will try to fetch 10 years worth of data
+
+    Syncing mechanism:
+
+    - use bookmark to keep track of the created_at
+    - sync records since the last bookmarked created_at
+    """
+
+    replication_keys = ["created_at"]
+    replication_method = "INCREMENTAL"
+
+    def get_default_start(self):
+        return (datetime.now() - timedelta(days=10*365)).strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    def get_params(self):
+        return {
+            "created_after": self.get_bookmark(),
+            "sort_by": "created_at",
+            "sort_order": "ascending",
```

Comment:
and how are we handling existing integer states?

Reply:
state for stream `visitor_activities` looks like 
```
"visitor_activities":{"id":"977882887"}
```

Meaning that, since it is a nested object with keys, there is no need to worry about the transition towards having `visitor_activities` as a created at stream. If we run an account with the example above, it will try to fetch the `created_at` field inside of `visitor_activities1; which, since it doesn't exist, will reddirect the stream to the default value (10 years ago from now).

Reply:
worth noticing that the `id` field will be kept in the state, so after a run, it will end up like
```
"visitor_activities": {"created_at": "2022-10-20 07:28:36", "id": "977882887"}
```